### PR TITLE
Add normalization transformation

### PR DESCRIPTION
**Pull request by lkuchenb**
This transformation normalizes the metadata model in the following way:

* Inherited slots and slot definitions are moved directly into the classes
* All inheritance relationships are removed
* The `types` schema is embedded directly into the schema if it is being imported.

### CODE REVIEW

File: metldata/builtin_transformations/normalize_model/main.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+# pylint: disable=unused-argument
+def check_model_assumptions(model: MetadataModel, config: NormalizationConfig):
+    """Check that the classes and slots specified in the config exist in the model."""
+
+
+# pylint: disable=unused-argument
+def transform_model(model: MetadataModel, config: NormalizationConfig) -> MetadataModel:
+    """Delete slots from classes in the model."""
```
Comment (Cito) [inline metldata/builtin_transformations/normalize_model/main.py:36]: Docstring needs to be adapted.

File: metldata/builtin_transformations/normalize_model/model_transform.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            raise RuntimeError("Types schema does not contain a dict of types.")
+        for prefix_name, prefix_def in types_schema.prefixes.items():
+            if prefix_name not in output_model["prefixes"]:
+                output_model["prefixes"][prefix_name] = prefix_def
+        # Drop import
+        output_model["imports"].remove("linkml:types")
+
+
+def normalize_model(model: MetadataModel) -> MetadataModel:
+    """Normalized model to canonical form with all slots being globally defined
```
Comment (Cito) [inline metldata/builtin_transformations/normalize_model/model_transform.py:69]:
```suggestion
    """Normalize model to canonical form with all slots being globally defined
```

**Review by Cito — APPROVED** [review]
Looks good. Just some docstrings/names should be adapted.

File: metldata/builtin_transformations/normalize_model/main.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        Args:
+            metadata: The metadata to be transformed.
+            annotation: The annotation on the metadata.
+
+        Raises:
+            MetadataTransformationError:
+                if the transformation fails.
+        """
+
+        return metadata
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply metldata/builtin_transformations/normalize_model/main.py:56]: Yes

**Review by Cito — APPROVED** [review]

**Pull request merged**